terraform/bigquery.tf

```diff
@@ -362,6 +461,61 @@ resource "google_bigquery_table" "unique_events_view" {
   ]
 }
 
+# Unique Events -deduplicate rows but as a table function
+resource "google_bigquery_routine" "unique_events_by_date_type" {
```

Comment:
Does this need `routine_type`, `language` and/or `return_table_type`?

Reply:
Good questions:

routineType: Probably shout set it

language: defaults to SQL unless you set remoteFunctionOption, so it should be fine, but I could add it

return_table_type: I left absent, though I think either way is fine. If absent, type is inferred at runtime, otherwise wit will cast to match the column types at query time. The former I think is less error prone, as any casting is avoided at the TVF level

Reply:
Ack, just wanted to clarify the TF docs didnt specify what the defaults were. Maybe add or document.

Reply:
Yeah, the TF documentation was a bit lacking, but the link to the API documentation did show what these do

Reply:
I added language and routineType explicitly, and left out return_table_type as its already well documented in the tf docs